tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -2904,6 +2935,11 @@ port::Status CudnnSupport::DoConvolve(
   switch (kind) {
     case dnn::ConvolutionKind::FORWARD: {
       SE_RETURN_IF_ERROR(get_fwd_bugs());
+      if (element_type == stream_executor::dnn::kInt8) {
```

Comment:
This is ugly. A solution to avoid this special case is to overload DoConvolve() to separate int8 from other types as what DoFusedConvolve (see the other caller to `IntgerConvolutionSanityChecks`) does. However, DoConvolve(int8) still need a if or switch statement to special case `dnn::ConvolutionKind::FORWARD`. Basically DoFusedConvolve only needs to dispatch for data types, while DoConvolve needs to double dispatch for both data types and convolution kinds. I am open for suggestions.

Reply:
I prefer the dispatch to be in a single function body, using `switch`es and `if`s. Multiple function signatures (through overloading or different function names) is very hard to maintain when the parameters change. I cleaned up most of them, but you can still see the leftovers in `stream.h`

Reply:
Since the sanity check is removed, this change is not necessary.